tests/test_cmdline_options.py

```diff
@@ -0,0 +1,199 @@
+
+
+def deps_option_strategy(paths: List[str]):
+    return st.lists(
+        st.sampled_from(paths),
+        min_size=0,
+        max_size=MAX_NUMER_OF_DEPS_ARGS,
+    ).map(lambda xs: ["--deps"] + xs if xs else [])
+
+
+ignored_strategy = st.lists(safe_string, min_size=0, max_size=MAX_IGNORE_ARGS)
+ignored_undeclared_strategy = ignored_strategy.map(
+    lambda xs: ["--ignore-undeclared"] + xs if xs else []
+)
+ignored_unused_strategy = ignored_strategy.map(
+    lambda xs: ["--ignore-unused"] + xs if xs else []
+)
+
+deps_parser_choice_strategy = st.one_of(st.sampled_from(deps_parser_choice), st.none())
+
+verbosity_indicator = st.text(
+    alphabet=["q", "v"], min_size=1, max_size=MAX_VERBOSITY_ARGS
+)
+verbosity_strategy = st.lists(
+    verbosity_indicator, min_size=0, max_size=MAX_VERBOSITY_ARGS
+).map(lambda xs: [f"-{x}" for x in xs])
+
+venv_strategy = st.one_of(st.sampled_from(venvs), st.none()).map(
+    lambda x: ["--pyenv", x] if x is not None else []
+)
+
+
+# ---------------------------------------------- #
+# Compose available strategies
+# ---------------------------------------------- #
+@st.composite
+def cli_arguments_combinations(draw):
+    to_stdin = None
+
+    project_dir = draw(
+        st.sampled_from([d for d in SAMPLE_PROJECTS_DIR.iterdir() if d.is_dir()])
+    )
+    code_paths = available_python_input(project_dir=project_dir)
+    dep_paths = available_deps(project_dir=project_dir)
+
+    # Dependencies are treated differently, as there is an explicit way
+    # to point to a parser that we want fawltydeps to use.
+    # Following code ensures that we do not encounter trying to match
+    # explicit file with incorrect parser (square in the oval-shaped hole problem)
+    drawn_deps = draw(deps_option_strategy(dep_paths))
+    deps_parser = []
+    drawn_deps_parser = draw(deps_parser_choice_strategy)
+    # a simple and not 100% accurate check if deps parser matches explicitly given file
+    if drawn_deps_parser is not None and all(
+        d.endswith(drawn_deps_parser) for d in drawn_deps
+    ):
+        deps_parser = ["--deps-parser-choice", drawn_deps_parser]
+
+    strategy = st.permutations(
+        [
+            draw(actions_strategy),
+            draw(output_formats_strategy),
+            drawn_deps,
+            draw(ignored_undeclared_strategy),
+            draw(ignored_unused_strategy),
+            deps_parser,
+            draw(verbosity_strategy),
+            # draw(venv_strategy), # TODO: better handle information on Python envs
+        ]
+    )
+    args = reduce(lambda x, y: x + y, draw(strategy))
+
+    # only `code` option changes to_stdin
+    code = draw(code_option_strategy(code_paths))
+    if code is not None:
+        if "-" in code:
+            to_stdin = example_python_stdin
+        args += code
+
+    return (args, to_stdin)
+
+
+@given(cli_arguments=cli_arguments_combinations())
+def test_options_interactions__correct_options__give_success_code(cli_arguments):
```

Comment:
Given that the assertion at the end of this function tests for `exit_code in {0, 3, 4}`, would it make sense to rename this to:

```suggestion
def test_options_interactions__correct_options__does_not_abort(cli_arguments):
```